react/features/app/functions.web.js

```diff
@@ -9,7 +16,35 @@ import JitsiMeetLogStorage from '../../../modules/util/JitsiMeetLogStorage';
 
 const Logger = require('jitsi-meet-logger');
 
-export * from './functions.native';
+export { _getRoomAndDomainFromUrlString } from './functions.native';
+
+/**
+ * Determines which route is to be rendered in order to depict a specific Redux
+ * store.
+ *
+ * @param {(Object|Function)} stateOrGetState - Redux state or Regux getState()
+ * method.
+ * @returns {Route}
+ */
+export function _getRouteToRender(stateOrGetState) {
+    const state
+        = typeof stateOrGetState === 'function'
+            ? stateOrGetState()
+            : stateOrGetState;
+
+    const { platform } = state['features/app'];
+    const { room } = state['features/base/conference'];
+    const { landingIsShown } = state['features/landing'];
+
+    let component = isRoomValid(room) ? Conference : WelcomePage;
+
+    // If landing was shown there is no need to show it again.
+    if (platform && !landingIsShown) {
+        component = detectAndroid() || detectIOS() ? Landing : component;
+    }
```

Comment:
I find the above `if` more complex than necessary:

1. There's an assignment of `component` to `component`.

2. There're 2 separate conditions that may be merged into 1 condition.